### PR TITLE
Added new route for profile plan type

### PR DESCRIPTION
**Pull request by jayeshmangwani**
### Fixed Issues

$ https://github.com/Expensify/App/issues/51768
PROPOSAL:

### Tests

1. Go to Workspaces -> Workspace -> Profile.
2. If the user has admin access, they will see the Plan Type tab under the Company address menu item.
3. Click on Plan Type to open a new RHP screen for Plan Type, where the current workspace plan will be preselected.

- [x] Verify that no errors appear in the JS console

### QA Steps

// TODO: These must be filled out, or the issue title must include "[No QA]."

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/670f30b6-10ec-4aa8-9073-703b02665008

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/b474a0a7-e58f-4ff2-94b1-7d2b12d8b81d

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/0a093048-18c7-458a-9590-5cabf5192b0b

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/a1e67340-cd74-4d85-a307-f52c0b984a46

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/cafeda4a-427d-47ed-8a16-977254be125a

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/aa13bbc3-c7f9-4d37-82d3-e17e9f797bfc

</details>

### CODE REVIEW

**Comment by jayeshmangwani** [comment]
In this PR, @DylanDylann , we’ve added the Plan Type row and a new RHP page specifically for Plan Type. Currently, we're only implementing the pre-selection of the Plan Type on this page.

**Comment by DylanDylann** [comment]
BUG: Offline indicator displays above the button

Screenshot 2024-11-12 at 11 30 52

**Comment by DylanDylann** [comment]
BUG: 
- The button should display above the device border
- The subtitle should be in the scroll view

https://github.com/user-attachments/assets/8e3a4139-1c18-4243-ae92-9d88816b27bb

**Comment by jayeshmangwani** [comment]
Thanks for the review, @DylanDylann! I’ve fixed the button position issue when offline. I’ll be adding the upgrade component to this PR next.
Simulator Screenshot - iPhone 15 Pro - 2024-11-12 at 18 46 22

**Comment by jayeshmangwani** [comment]
> The subtitle should be in the scroll view

We’re currently keeping the subtitle text outside the ScrollView for the SelectionList on some pages, like the Priority and Theme pages. I don’t have a strong opinion on this; I’ve kept it outside mainly for consistency.

https://github.com/user-attachments/assets/f09d70d7-2a30-41cc-881b-8e3465f4b28a

**Comment by DylanDylann** [comment]
@jayeshmangwani Thanks for your update. Please ping me when you push the new code for upgrade component

File: src/CONST.ts
Comment (DylanDylann) [inline src/CONST.ts:934]: Add my review

Comment (jayeshmangwani) [reply src/CONST.ts:934]: Used URL as per this comment https://github.com/Expensify/App/issues/51494#issuecomment-2462110622

File: src/ROUTES.ts
Comment (jayeshmangwani) [inline src/ROUTES.ts:974]: We're using the existing Upgrade page to display the upgrade workspace page when coming from the workspace plan page. Now, we can also navigate from pages that don't have a specific feature.

I've made featureName an optional parameter. If it's not provided, we'll show the common control plan features instead.

File: src/languages/en.ts
Comment (jayeshmangwani) [inline src/languages/en.ts:4377]: these translations confirmed here https://expensify.slack.com/archives/C01GTK53T8Q/p1731048615013379?thread_ts=1730105986.836259&cid=C01GTK53T8Q

File: src/languages/en.ts
Comment (jayeshmangwani) [inline src/languages/en.ts:4257]: These translations are yet to be confirmed here: https://expensify.slack.com/archives/C01GTK53T8Q/p1731048615013379?thread_ts=1730105986.836259&cid=C01GTK53T8Q

File: src/pages/workspace/WorkspaceProfilePlanTypePage.tsx
Comment (jayeshmangwani) [inline src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:55]: `isPlanTypeLocked` is not the final logic here, as I'm currently unsure how to determine if a specific policy is on a yearly plan. I've asked for clarification https://expensify.slack.com/archives/C02NK2DQWUX/p1731682677445059, but for now, we can finalize the UI and adjust the show logic later.

**Comment by jayeshmangwani** [comment]
**Plan Types  and Workspace Upgrade Attachments**




<details>
  <summary>iOS</summary>

  https://github.com/user-attachments/assets/0a82dfb8-42c8-47f9-b4ef-b5b47d2f3115

 
</details>


<details>
  <summary>Desktop:</summary>
  
  https://github.com/user-attachments/assets/92cb7383-557f-4a10-80a7-d46a68fe3626

  
</details>

<details>
  <summary>web:</summary>
  
  https://github.com/user-attachments/assets/33337af7-0035-47bd-a093-99649c74e00d

  
</details>

**Comment by jayeshmangwani** [comment]
@DylanDylann I've added the Plan Type page and handled the upgrade workspace case. Feel free to review! A few things are yet to be confirmed (I’ve added comments above for those), but they shouldn't block the initial review of these changes.

Thanks!

File: src/pages/workspace/upgrade/UpgradeIntro.tsx
Comment (DylanDylann) [inline src/pages/workspace/upgrade/UpgradeIntro.tsx:101]: @jayeshmangwani Let's move this component to a new file

File: src/CONST.ts
Comment (DylanDylann) [reply src/CONST.ts:934]: No, my comment is to add this PR to my K2s List. No problem here

**Comment by DylanDylann** [comment]
@jayeshmangwani Also please help to resolve conflict

**Comment by DylanDylann** [comment]
Please add pending action to the new field when creating a new workspace 

https://github.com/user-attachments/assets/317d69ca-04ab-4bda-85b8-2fa68a291a14

**Comment by jayeshmangwani** [comment]
Thanks for the review, @DylanDylann! I’ve resolved the merge conflicts and `CommonCorporateFeatures` migrated to a new component (GenericFeaturesView). However, I’m not sure if the new component name fully justifies its purpose 😄.

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by DylanDylann — APPROVED** [review]

**Comment by DylanDylann** [comment]
@carlosmiceli Friendly bump

**Comment by carlosmiceli** [comment]
Thanks for the bump, been waiting for a review on a BE PR to get merged. @DylanDylann correct me if I'm wrong, but wouldn't we risk showing a non-working plan type selector with this PR if the BE is not ready?

**Comment by DylanDylann** [comment]
@carlosmiceli We can merge this PR first without an update from BE, this PR is about the upgrade flow and the API for this is done before

**Comment by carlosmiceli** [comment]
Gotcha, let's solve conflicts and we can merge! 🙇 
cc @jayeshmangwani

File: src/languages/en.ts
Comment (jayeshmangwani) [reply src/languages/en.ts:4275]: @carlosmiceli Before merging, I wanted to let you know that these translations are not yet confirmed.

File: src/pages/workspace/WorkspaceProfilePlanTypePage.tsx
Comment (jayeshmangwani) [reply src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:55]: Before merging, I also wanted to let you know that `isPlanTypeLocked` logic is also not yet confirmed

Comment (jayeshmangwani) [reply src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:55]: @carlosmiceli ⬆️

**Comment by jayeshmangwani** [comment]
Conflicts have been resolved @carlosmiceli. Before merging the PR, please check these comments: https://github.com/Expensify/App/pull/52117#discussion_r1860918653 and https://github.com/Expensify/App/pull/52117#discussion_r1860919403

File: src/pages/workspace/WorkspaceProfilePlanTypePage.tsx
Comment (carlosmiceli) [reply src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:55]: Got it, if we can add it later then that's all good.

File: src/languages/en.ts
Comment (carlosmiceli) [reply src/languages/en.ts:4286]: Great, let's keep these comments open so we can go back to them to see what we have yet to change/merge.

File: src/languages/en.ts
Comment (carlosmiceli) [reply src/languages/en.ts:4406]: Can we resolve this conversation?

**Review by carlosmiceli — APPROVED** [review]

File: src/pages/workspace/WorkspaceProfilePlanTypePage.tsx
Comment (carlosmiceli) [reply src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:55]: I think there's a few relevant answers now in the slack thread :)

**Comment by carlosmiceli** [comment]
@jayeshmangwani how does it work with the changes from this PR that got reverted? Are you creating another PR with the same changes now that the BE is ready?

**Comment by jayeshmangwani** [comment]
Yes, I'll be raising the PR in a few hours. I'm away from the keyboard right now.

**Comment by carlosmiceli** [comment]
No rush at all, it's Saturday 😄

**Comment by jayeshmangwani** [comment]
@carlosmiceli @DylanDylann New PR containing the same changes has been created here https://github.com/Expensify/App/pull/53343

**Pull request merged**
